Send insights from categories filter. Implements #404

dmfs opened the pull request.

dmfs committed:
Send insights from categories filter. Implements #404

lemonboston commented on eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/filter/insights/Opened.java:
```diff
@@ -0,0 +1,61 @@
+    private final FilterStep mDelegate;
+
+
+    public Opened(FilterStep delegate)
```
Maybe have default ctor which creates the `FilterStep` for this class and all others to have shorter usage in `EventFilterView`? I know these are decorators and with those ctors they wouldn't be but it would be fine imo.

dmfs replied:
Good idea. Done.

lemonboston commented on eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/filter/views/EventFilterView.java:
```diff
@@ -229,6 +236,11 @@ public void onClick(CategoryOption categoryOption)
             mShowAllItemView.update(mFilterState);
 
+            new InsightsTask(getContext()).execute(
+                    categoryOption.isSelected() ?
```
Maybe have something like `new Tapped(categoryOption)` to hide all this?

dmfs replied:
right, I replaced it with `Toggled`

lemonboston reviewed (approved):
Code looks correct and the design of these Steps is nice I think. I didn't debug the actual jsons, so I haven't double checked that.

I added comments about possibly making the usage shorter, but of course those are not critical.

dmfs committed:
Add secondary ctors
Add Toggled as a replacement for Selected and Unselected (keeping the latter two for future use cases)
Add requets body logging to RequestUriAndTimeLogging

dmfs commented:
@lemonboston please check again

lemonboston commented:
The changes for the `Step`s look good, although `Toggled` could also just delegate to `Selected`, `Unselected` if already available. (Would reduce duplicate definition of path strings "selected", "unselected")

I think the request body logging in the executor decorator could use `Log.d()` instead of `System.out` and moved after the request line log, before the response time log. And probably only logged when content length is > 0. Message could start with the `(4)` like id like the other two, sg like `(4) Request body: ..`.

dmfs committed:
improve request logging

dmfs committed:
Let Toggled delegate to Selected and Unselected

dmfs commented:
Ok, I've addressed your comments.
